popup/searchResults.js

```diff
@@ -0,0 +1,251 @@
+      }
+    */
+
+    // TODO: Check if search result is already installed.
+    //       If so hide it, or mark as installed with an "Uninstall" button.
+
+    const entry = template.searchResult.cloneNode(true);
+    Object.assign(entry, {
+      id: ENTRY_ID_PREFIX_RAW + userstyleSearchResult.id,
+      styleId: userstyleSearchResult.id
+    });
+    $('#searchResults-list').appendChild(entry);
+
+    const title = $('.searchResult-title', entry);
+    Object.assign(title, {
+      textContent: userstyleSearchResult.name,
+      title: userstyleSearchResult.name,
+      href: 'https://userstyles.org' + userstyleSearchResult.url,
+      onclick: handleEvent.openURLandHide
+    });
+
+    const screenshot = $('.searchResult-screenshot', entry);
+    let ss_url = userstyleSearchResult.screenshot_url;
+    if (RegExp(/^[0-9]*_after.(jpe?g|png|gif)$/i).test(ss_url)) {
+      ss_url = 'https://userstyles.org/style_screenshot_thumbnails/' + ss_url;
+    }
+    Object.assign(screenshot, {
+      src: ss_url,
+      title: userstyleSearchResult.name
+    });
+
+    // TODO: Expand/collapse description
+    const description = $('.searchResult-description', entry);
+    Object.assign(description, {
+      textContent: userstyleSearchResult.description.replace(/<.*?>/g, ""),
+      title: userstyleSearchResult.description.replace(/<.*?>/g, "")
+    });
+
+    const authorLink = $('.searchResult-authorLink', entry);
+    Object.assign(authorLink, {
+      textContent: userstyleSearchResult.user.name,
+      title: userstyleSearchResult.user.name,
+      href: 'https://userstyles.org/users/' + userstyleSearchResult.user.id,
+      onclick: handleEvent.openURLandHide
+    });
+
+    // TODO: Total & Weekly Install Counts
+    // TODO: Rating
+
+    const installButton = $('.searchResult-install', entry);
+    const name = userstyleSearchResult.name;
+    Object.assign(installButton, {
+      onclick: install
+    });
+
+    /** Installs the current userstyleSearchResult into stylus. */
+    function install(event) {
+      UserStylesAPI.fetch("/api/v1/styles/" + userstyleSearchResult.id)
+        .then(styleObject => {
+          console.log("TODO: Install style ID", userstyleSearchResult.id);
+          console.log("Full styleObject:", styleObject);
+          /*
+           * FIXME
+           * Sample full styleObject: https://userstyles.org/api/v1/styles/70271
+           * We need to convert this sytleObject into the format expected by saveStyleSafe
+           * I.e. styleObject.id is the ID of the userstyles.org style (e.g. 70271 above)
+           */
+
+          // messaging.js#saveStyleSafe({...}) expects an "id" referring to the Stylus ID (1-n).
+          delete styleObject.id;
+
+          Object.assign(styleObject, {
+            // TODO: Massage styleObject into the format expected by saveStyleSafe
+            enabled: true,
+            reason: 'update',
+            notify: true
+          });
+          saveStyleSafe(styleObject);
```

Comment:
Here's where I (incorrectly) try to install the style. Comments explain my reasoning: I'm trying to pass a StyleObject from userstyles.org's API (it has [a specific format](https://userstyles.org/api/v1/styles/70271)) into the `saveStyleSafe` method, which expects some other format for input.

Reply:
Direct saving requires sectioned JSON which we fetch from links like `https://userstyles.org/styles/chrome/37035.json` or `https://userstyles.org/styles/chrome/37035.json?ik-syntax-theme=ik-Twilight&ik-syntax-codemirror=ik-cm-twilight&ik-syntax-jupyter=ik-jp-twilight&ik-bg-choice=ik-yes&ik-bg-options=ik-Tiled&ik-bg-attachment=ik-Scroll&ik-tab-size=ik-4&ik-code-wrap=ik-nowrap&ik-base-color=%234183C4&ik-font-choice=Menlo&ik-bg-url=ik-default` for customizable styles.

In your link it's a raw CSS string that needs to be converted with our `moz-parser.js` (if you go this route, load it dynamically with `loadScript`). However you'll need to render the customizable variables e.g. `/*[[base-color]]*/` in [github dark](https://userstyles.org/styles/37035/github-dark). And we'll need to allow users to customize those variables (for example the way we do with usercss in the style manager) or at least tell them to do it on the style page.